### PR TITLE
HAI-1944 Translations for emails

### PR DESCRIPTION
**Pull request by corvidian**
# Description

Add translations for the hanke invitation and hakemus notification.

Pull the signatures to separate partials, since they are big identical blocks between the two emails. They also need to be translated just once.

Test the translations in a unit test to make it easier.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [X] Other

# Instructions for testing
Create a johtoselvityshakemus and send it. Check smtp4dev for the emails (http://localhost:3003/).

# Checklist:

- [X] I have written new tests (if applicable)
- [X] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
hanke-email
hakemus-email

### CODE REVIEW

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/email/EmailSenderServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+                assertThat(htmlBody)
+                    .contains("""<a href="https://haitaton.hel.fi/fi/kutsu?id=$invitationToken">""")
+            }
+
+            @Test
+            fun `contains the signature`() {
+                val (textBody, htmlBody) = sendAndCapture().bodies()
+
+                assertThat(textBody).all {
+                    containsLine("Tämä on automaattinen sähköposti – älä vastaa tähän viestiin.")
```
Comment (pitkni) [inline services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/email/EmailSenderServiceTest.kt:121]: Maybe something could be done about the code quality report (duplication)? I like the explicit style here, but it does make the sonar report look worse than it actually is. I will leave this up to your consideration.

Comment (corvidian) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/email/EmailSenderServiceTest.kt]: >Programs must be written for people to read, and only incidentally for machines to execute.
― Harold Abelson, Structure and Interpretation of Computer Programs

I do believe this quote is on to something. And I think static code analysis falls under the "machine" category.

Still, I refactored the tests to have less duplication in them. See how you like them.

Comment (pitkni) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/email/EmailSenderServiceTest.kt]: > Programs must be written for people to read (and maintain).
-- Me

This made sonar happy also. 👍

File: services/hanke-service/build.gradle.kts [hunk 2]
```diff
@@ -165,3 +180,17 @@ tasks.register("installGitHook", Copy::class) {
 tasks.named("build") { dependsOn(tasks.named("installGitHook")) }
 
 tasks.named("check") { dependsOn(tasks.named("integrationTest")) }
+
+tasks.named("processResources") { dependsOn(tasks.named("copyEmailTemplates")) }
+
+mjml {
```
Comment (pitkni) [inline services/hanke-service/build.gradle.kts:186]: The plugin thing is a very good improvement.

However, should this part have some comments on what these mean? 

For example, I have no idea what `juiceOptions.set("{\"preserveImportant\": true}")` does.

Comment (corvidian) [reply services/hanke-service/build.gradle.kts:186]: I added a link to MJML documentation where the config options are explained. Also, a separate link to html-minifier documentation for the minifyOptions.

The juice-stuff didn't do anything for us, so I removed it. It was copied along an example config.

**Review by pitkni — APPROVED** [review]

**Pull request merged**